veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,228 @@
+
+Similarly to the CPU and memory hotplug features, a VMI that has the `HotVNICChange` condition set to `true`:
+1. Shall be considered for automatic migration.
+2. Shall not be considered for migration abortion.
+
+#### Pros
+
+- Improves the current user experience
+- Will work for hotplugging interfaces using bridge and SR-IOV bindings
+- Keeps the integration with the dynamic-networks-controller
+
+#### Cons
+ - Will not work for non-migratable VMs
+ - Performs a migration for every hotplug / hotunplug
+ - Requires an API change
+
+## API Examples
+
+<!--
+Tangible API examples used for discussion
+-->
+
+## Alternatives
+
+### Using the dynamic-networks-controller
+
+The dynamic-networks-controller is a controller, running as a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/),
+listening to changes on Pods and invoking the [Multus v4 CNI](https://github.com/k8snetworkplumbingwg/multus-cni).
+
+It enables NIC hotplug / hotunplug operations in place, e.g. without creating another pod.
+A known limitation of the controller above is that it cannot hotplug interfaces that require invocation of
+[Kubernetes device plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/),
+for example: SR-IOV network interfaces.
+
+#### Pros
+
+- Improved user experience
+- No need for migration when attaching/detaching interfaces using bridge binding
+- Works on non-migratable VMs
+- Does not require changes to kubevirt/kubevirt
+
+#### Cons
+
+- The dynamic-networks-controller is an external component that needs to be deployed and managed
+- Not applicable for hotplugging SR-IOV interfaces
+
+### Providing Namespace Admins the required RBAC Rules
+
+Cluster admins could provide Namespace admins the permissions to create VMIM objects.
+The documentation will be updated to reflect the fact that there are additional RBAC rules missing for namespace
+admins to be able to perform NIC hotplug / unplug.
+
+#### Pros
+
+- Does not require code changes
+
+#### Cons
+
+- Losses the benefits of limiting VMIM creation to cluster admins
+- Requires cluster-admins to take action
+
+## Scalability
+
+N/A.
+
+## Update/Rollback Compatibility
+
+Users will have to configure their KubeVirt configuration similarly to CPU and memory hotplug in order to use the automatic
+migration.
+The manual migration will no longer achieve the desired affect.
+Users using the dynamic-networks-controller will have to adjust their KubeVirt configuration.
+
+## Functional Testing Approach
+
+The migration-based hotplug / hotunplug tests will be adjusted to remove the manual VMIM object creation.
+The in-place hotplug / hotunplug tests will be adjusted to assert that a migration did not occur.
+
+## Implementation Phases
+
+1. Support automatic migration on NIC hotplug / hotunplug (without dynamic-networks-controller).
+2. Support in-place NIC hotplug / hotunplug (with dynamic-networks-controller).
```

Comment:
I do not understand this one. It is an existing capability, so how can you consider it as an addition when it is already there?

Reply:
This phase was removed.